flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkInputFormat.java

```diff
@@ -104,6 +105,22 @@ public void testNestedProjection() throws Exception {
   }
 
   private List<Row> runFormat(FlinkInputFormat inputFormat) throws IOException {
+    return getRows(inputFormat);
+  }
+
+  public static List<RowData> getRowData(FlinkInputFormat inputFormat) throws IOException {
+    RowType rowType = FlinkSchemaUtil.convert(inputFormat.projectedSchema());
+
+    DataStructureConverter<Object, Object> converter = DataStructureConverters.getConverter(
+        TypeConversions.fromLogicalToDataType(rowType));
+
+    return getRows(inputFormat).stream()
+        .map(converter::toInternal)
+        .map(RowData.class::cast)
+        .collect(Collectors.toList());
+  }
+
+  public static List<Row> getRows(FlinkInputFormat inputFormat) throws IOException {
```

Comment:
I think it is a bad practice to make helper methods in one test suite public and use them in another suite. Instead, helper methods should be moved to an appropriate test utility class. That way, we don't have test utility code that is hard to find because it lives in whatever test was written first.

Reply:
OK, I added `TestHelpers` to contain `getRows` and `getRowData`.